File: in_Java/0232 Implement Queue using Stacks.java
class MyQueue {

    Stack<Integer> st1;
    Stack<Integer> st2;
    /** Initialize your data structure here. */
    public MyQueue() {
        st1 = new Stack<>();
        st2 = new Stack<>();
    }
    
    /** Push element x to the back of queue. */
    public void push(int x) {
        st1.push(x);
    }
    
    /** Removes the element from in front of queue and returns that element. */
    public int pop() {
        while(!st1.isEmpty()){
            st2.push(st1.pop());
        }
        int res = st2.pop();
        while(!st2.isEmpty()) st1.push(st2.pop());
        return res;
    }
    
    /** Get the front element. */
    public int peek() {
        while(!st1.isEmpty()){
            st2.push(st1.pop());
        }
        int ress = st2.peek();
        while(!st2.isEmpty()) st1.push(st2.pop());
        return ress;
    }
    
    /** Returns whether the queue is empty. */
    public boolean empty() {
        return st1.isEmpty() && st2.isEmpty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */

File: in_Python/0232 Implement Queue using Stacks.py
class MyQueue(object):

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.st1 = []
        self.st2 = []
        

    def push(self, x):
        """
        Push element x to the back of queue.
        :type x: int
        :rtype: None
        """
        self.st1.append(x)

    def pop(self):
        """
        Removes the element from in front of queue and returns that element.
        :rtype: int
        """
        while len(self.st1) != 0:
            self.st2.append(self.st1.pop())
        a = self.st2.pop()
        while len(self.st2) != 0:
            self.st1.append(self.st2.pop())
        return a
    def peek(self):
        """
        Get the front element.
        :rtype: int
        """
        while len(self.st1) != 0:
            self.st2.append(self.st1.pop())
        b = self.st2[-1]
        while len(self.st2) != 0:
            self.st1.append(self.st2.pop())
        return b
    def empty(self):
        """
        Returns whether the queue is empty.
        :rtype: bool
        """
        return len(self.st1) == 0 
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()